external/buildscripts/Build.bee.cs

```diff
@@ -1,133 +1,81 @@
-					"testing"));
-
-			Artifacts.Add("toolchain-llvm-centos",
-				new Tuple<string, string>(
-					"toolchain-llvm-centos/9.0.1-7.7.1908-v0_9c1c26d5205c7ead3cba5a545796579d753c459ac5e16d0690d1b087efa7835d.7z",
-					"testing"));
-		}
-
-		private static void RegisterCommonNonWindowsArtifacts()
-		{
-			Artifacts.Add("libtool-src",
-				new Tuple<string, string>(
-					"libtool-src/2.4.6_49a0ed204b3b24572e044400cd05513f611bcca6ced0d0816a57ac3b17376257.7z",
-					"public"));
-
-			Artifacts.Add("texinfo-src",
-				new Tuple<string, string>(
-					"texinfo-src/4.8_975b9657ebef8a4fe3897047ca450b757a0a956b05399dc813f63e84829bac6a.7z",
-					"public"));
-
-			Artifacts.Add("automake-src",
-				new Tuple<string, string>(
-					"automake-src/1.16.1_d281b950e26265f55f0a63188a8c6388e638b354b7ed80d186690119cbc4f953.7z",
-					"public"));
-
-			Artifacts.Add("autoconf-src",
-				new Tuple<string, string>(
-					"autoconf-src/2.69_0e4ba7a0363c68ad08a7d138b228596aecdaea68e1d8b8eefc645e6ac8fc85c7.7z",
-					"public"));
-
-			Artifacts.Add("libgdiplus-mac",
-				new Tuple<string, string>(
-					"libgdiplus-mac/9df1e3b3b120_4cf7c08770db93922f54f38d2461b9122cddc898db58585864446e70c5ad3057.7z",
-					"unity-internal"));
-		}
-	}
+    public class BuildProgram
+    {
+        private static readonly List<StevedoreArtifact> StevedoreArtifacts = new List<StevedoreArtifact>();
+
+        internal static void Main()
+        {
+            var manifest = "manifest.stevedore";
+
+            Backend.Current.StevedoreSettings = new StevedoreSettings
+            {
+                Manifest = { manifest }
+            };
+
+            RegisterCommonArtifacts();
+
+            if (Platform.HostPlatform is WindowsPlatform)
+            {
+                RegisterWindowsArtifacts();
+            }
+            else
+            {
+                RegisterCommonNonWindowsArtifacts();
+
+                if (Platform.HostPlatform is MacOSXPlatform)
+                {
+                    RegisterOSXArtifacts();
+                }
+                else if (Platform.HostPlatform is LinuxPlatform)
+                {
+                    RegisterLinuxArtifacts();
+                }
+            }
+
+            foreach (var artifact in StevedoreArtifacts)
+            {
+                Console.WriteLine($">>> Registering artifact {artifact.ArtifactName}");
+                artifact.UnpackToUnusualLocation($"artifacts/Stevedore/{artifact.ArtifactName}");
+            }
+        }
+
+        private static void RegisterCommonArtifacts()
+        {
+            StevedoreArtifacts.Add(new StevedoreArtifact("MonoBleedingEdge"));
+            StevedoreArtifacts.Add(new StevedoreArtifact("reference-assemblies"));
+        }
+
+        private static void RegisterWindowsArtifacts()
+        {
+            StevedoreArtifacts.Add(new StevedoreArtifact("android-ndk-windows-x86_64"));
```

Comment:
Does this work? the name here does not match the name in the manifest `android-ndk-win`

We have to check/update the build scripts if the ndk directory name has changed

Reply:
It does not I am fixing this currently.

Reply:
the buildscript needs updated here https://github.com/Unity-Technologies/mono/blob/unity-master/external/buildscripts/build.pl#L671

Reply:
Already was updated earlier (note this PR is for mono upgrade) https://github.com/Unity-Technologies/mono/blob/unity-master-new-unitychanges/external/buildscripts/build.pl#L673